dev/com.ibm.ws.persistence_fat/test-applications/consumer/src/persistence_fat/consumer/web/ConsumerServlet.java

```diff
@@ -0,0 +1,342 @@
+        Person p = bean.findPersonById(personId);
+        if (p == null) {
+            throw new RuntimeException("didn't find person with id " + personId);
+        }
+        String found = p.getData();
+        if (found != null) {
+            throw new RuntimeException("Expected person(" + personId + ") to have data(" + "null" + "), but found ("
+                                       + found + ").");
+        }
+    }
+
+    public void testPersistEmptyString() throws Exception {
+        _ut.begin();
+        long personId = consumer.persistPerson(false, 10, "");
+        _ut.commit();
+        // Use jpa to find person, validate String data is expected
+        Person p = bean.findPersonById(personId);
+        if (p == null) {
+            throw new RuntimeException("didn't find person with id " + personId);
+        }
+        String found = p.getData();
+        if (found != "") {
+            throw new RuntimeException("Expected person(" + personId + ") to have data(" + "{empty}"
+                                       + "), but found (" + found + ").");
+        }
+    }
+
+    // http://unicode-table.com/en/#0275
+    @Test
+    public void testQueryInvalidStrig() {
+        try {
+            String str = "\u0274\u0275";
+            consumer.queryPersonDataLiteral("\u0274\u0275");
+            throw new RuntimeException("Shouldn't have been able to execute a query with " + str + " in it.");
+        } catch (RuntimeException e) {
+            // expected
+        }
+
+        try {
+            String str = "\u0274\u0275";
+            consumer.queryPersonDataParameter("\u0274\u0275");
+            throw new RuntimeException("Shouldn't have been able to execute a query with " + str + " in it.");
+        } catch (RuntimeException e) {
+            // expected
+        }
+    }
+
+    @Test
+    public String testGetPersonName() throws Exception {
+        _ut.begin();
+        long personId = consumer.persistPerson(false, 10);
+        _ut.commit();
+
+        String[] psNames = consumer.getPersonName(personId);
+        if (psNames[0] == null || psNames[1] == null) {
+            throw new Exception("Found a null name. " + Arrays.toString(psNames));
+        }
+        if (!psNames[0].equals("first") || !psNames[1].equals("last")) {
+            throw new Exception("expected first, last but found " + psNames[0] + ", " + psNames[1]);
+        }
+        Person jpaPerson = bean.findPersonById(personId);
+        String[] jpaNames = new String[] { jpaPerson.getFirst(), jpaPerson.getLast() };
+
+        if (!Arrays.equals(jpaNames, psNames)) {
+            throw new Exception("Person names didn't match up. Expected " + Arrays.toString(jpaNames)
+                                + " but found " + Arrays.toString(psNames));
+        }
+        return Arrays.toString(psNames);
+    }
+
+    // @Test -- this recreates defect 159874
+    public void testSerializable() throws Exception {
```

Comment:
should this test be enabled now?

Reply:
Possibly deactivated test, left alone for now.